merlin/dataloader/torch.py

```diff
@@ -144,11 +141,7 @@ def _pull_values_offsets(self, values_offset):
         else:
             values = values_offset.flatten()
             offsets = torch.arange(values.size()[0], device=self.device)
-        num_rows = len(offsets)
-        if HAS_GPU:
-            offsets = torch.cat([offsets, torch.cuda.LongTensor([len(values)], device=self.device)])
-        else:
-            offsets = torch.cat([offsets, torch.LongTensor([len(values)])])
```

Comment:
I remember there was some issue with NVTabular and/or T4Rec in the multi-GPU case and `device=self.device` was added to line 149. What do you think about still keeping something like `if HAS_GPU: offsets = offsets.to(device=self.device)`?

Reply:
On second thought, it was probably an issue because of `torch.cat` and `offsets` and the other tensor being on different devices, so it should be okay with the refactor.

Reply:
the reason for the removal of the lines doing torch.cat is because the offsets previously were missing the last value. After this change the offsets are now computed in the `_row_lengths_to_offests` method which doesn't require this logic. There's still a torch.cat in use in that method, that concatenates the cumulative sum of the row lengths with a leading zero.